internal/common/function/function.go

```diff
@@ -219,6 +220,32 @@ func ValidateBslSpec(ctx context.Context, clientInstance client.Client, nonAdmin
 	} else if nonAdminBsl.Spec.BackupStorageLocationSpec.Credential.Name == constant.EmptyString || nonAdminBsl.Spec.BackupStorageLocationSpec.Credential.Key == constant.EmptyString {
 		return fmt.Errorf("NonAdminBackupStorageLocation spec.bslSpec.credential.name or spec.bslSpec.credential.key is not set")
 	}
+	bslSyncPeriodErrorMessage := "NABSL spec.backupStorageLocationSpec.backupSyncPeriod (%v) can not be greater or equal non admin backupSyncPeriod (%v)"
+	if nonAdminBsl.Spec.BackupStorageLocationSpec.BackupSyncPeriod != nil {
+		if appliedBackupSyncPeriod <= nonAdminBsl.Spec.BackupStorageLocationSpec.BackupSyncPeriod.Duration {
+			return fmt.Errorf(
+				bslSyncPeriodErrorMessage,
+				nonAdminBsl.Spec.BackupStorageLocationSpec.BackupSyncPeriod.Duration, appliedBackupSyncPeriod,
+			)
+		}
+	} else {
+		if defaultBackupSyncPeriod != nil {
+			if appliedBackupSyncPeriod <= *defaultBackupSyncPeriod {
+				return fmt.Errorf(
+					bslSyncPeriodErrorMessage,
+					defaultBackupSyncPeriod, appliedBackupSyncPeriod,
```

Comment:
Is this right? The `defaultBackupSyncPeriod` is a pointer and the `appliedBackupSyncPeriod` is not, so should this be used as pointer `*defaultBackupSyncPeriod` ?

Reply:
I think is ok, Go renders message correctly
```
NABSL spec.backupStorageLocationSpec.backupSyncPeriod (1h40m0s) can not be greater or equal non admin backupSyncPeriod (30s)
```